### PR TITLE
Fix #4446 - add SizingSystem::autosizedOccupantDiversity.

### LINKED ISSUE #4446: E+ 9.6.0: Add `autosizedOccupantDiversity` for SizingSystem
Enhancement Request
--------------
Get autosized occupant diversity value from somewhere in the sql file.

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #4446 

As indicated on  https://github.com/NREL/OpenStudio/issues/4446#issuecomment-930264664 I got puzzled by the fact that the tabular reports the Occupant Diversity for both heating and cooling.

But both report the same value:

https://github.com/NREL/EnergyPlus/blob/f420c06a69737dddcf4ba740ad855b45a03a8ef3/src/EnergyPlus/SizingManager.cc#L1594-L1595

```
        OutputReportPredefined::PreDefTableEntry(
            state, state.dataOutRptPredefined->pdchS62svrClD, FinalSysSizing(AirLoopNum).AirPriLoopName, state.dataSize->DBySys(AirLoopNum), 4); // D
```

https://github.com/NREL/EnergyPlus/blob/f420c06a69737dddcf4ba740ad855b45a03a8ef3/src/EnergyPlus/SizingManager.cc#L1679-L1680

```
        OutputReportPredefined::PreDefTableEntry(
            state, state.dataOutRptPredefined->pdchS62svrHtD, FinalSysSizing(AirLoopNum).AirPriLoopName, state.dataSize->DBySys(AirLoopNum), 4); // D
```

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/SizingSystem.cpp [hunk 1]
```diff
@@ -953,6 +953,45 @@ namespace model {
       return result;
     }
 
+    boost::optional<double> SizingSystem_Impl::autosizedOccupantDiversity() const {
+      boost::optional<double> result;
+
+      // Note: as of 9.6.0, for some reason, it returns one entry for cooling and one for heating...
+      // And it's only present in the tabular report 'Standard62.1Summary'
+      // Both the Heating and Cooling are actually the same underlying value. Cf https://github.com/NREL/OpenStudio/pull/4450#issue-1011104323
+      std::string tableName = "System Ventilation Requirements for Cooling";
```
Comment (jmarrec) [inline src/model/SizingSystem.cpp:962]: Note: I'm using the Cooling one, but both are exactly the same

File: src/model/SizingSystem.cpp [hunk 2]
```diff
@@ -982,6 +1021,11 @@ namespace model {
       if (val) {
         setCentralHeatingMaximumSystemAirFlowRatio(val.get());
       }
+
+      val = autosizedOccupantDiversity();
+      if (val) {
+        setOccupantDiversity(val.get());
+      }
```
Comment (jmarrec) [inline src/model/SizingSystem.cpp:1028]: Add it to applySizingValues() too

**Review by jmarrec — COMMENTED** [review]
I manually tested this by using this executable on the baseline_sys07.rb resources test.

cd testruns/baseline_sys07.rb/run/

```ruby
[1] run(main)> m = osload('in.osm')
sqlF=> #<OpenStudio::Model::Model:0x0000563c932ff910 @__swigtype__="_p_openstudio__model__Model">
[2] run(main)> sqlFile = OpenStudio::SqlFile.new("eplusout.sql")
=> #<OpenStudio::SqlFile:0x0000563c92eaa6a8 @__swigtype__="_p_openstudio__SqlFile">
[3] run(main)> m.setSqlFile(sqlFile)
=> true
[4] run(main)> a = m.getAirLoopHVACs[0]
=> #<OpenStudio::Model::AirLoopHVAC:0x0000563c946be7f8 @__swigtype__="_p_openstudio__model__AirLoopHVAC">
[5] run(main)> sz = a.sizingSystem
=> #<OpenStudio::Model::SizingSystem:0x0000563c946159f0 @__swigtype__="_p_openstudio__model__SizingSystem">
[6] run(main)> sz.autosizedOccupantDiversity
=> #<OpenStudio::OptionalDouble:0x0000563c945b4e48 @__swigtype__="_p_boost__optionalT_double_t">
[7] run(main)> sz.autosizedOccupantDiversity.get
=> 1.0

```

**Review by tijcolem — APPROVED** [review]
Looks good and all green!

**Pull request merged**